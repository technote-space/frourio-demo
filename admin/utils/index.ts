import { singular, plural } from 'pluralize';

export const getWord = (word: string, number: number): string => number <= 1 ? singular(word) : plural(word);
