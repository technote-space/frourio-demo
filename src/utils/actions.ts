import type { UseToastOptions } from '@chakra-ui/react';
import type { PageKeys } from '~/_pages';

type Dispatch = ((value: { type: string; [key: string]: any }) => void);

export const setAdmin = (dispatch: Dispatch, admin: { name?: string; icon?: string }) => dispatch({
  type: 'SET_ADMIN',
  admin,
});

export const openSidebar  = (dispatch: Dispatch) => dispatch({ type: 'OPEN_SIDEBAR' });
export const closeSidebar = (dispatch: Dispatch) => dispatch({ type: 'CLOSE_SIDEBAR' });

export const changePage = (dispatch: Dispatch, page: PageKeys) => dispatch({ type: 'PAGE', page });
export const logout     = (dispatch: Dispatch) => dispatch({ type: 'LOGOUT' });

export const createToast        = (dispatch: Dispatch, option: UseToastOptions) => dispatch({
  type: 'TOAST',
  toasts: option,
});
export const createWarningToast = (dispatch: Dispatch, option: UseToastOptions) => createToast(dispatch, {
  ...option,
  status: 'warning',
});
export const createErrorToast   = (dispatch: Dispatch, option: UseToastOptions) => createToast(dispatch, {
  ...option,
  status: 'error',
});
