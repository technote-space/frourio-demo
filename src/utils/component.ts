import { FC } from 'react';
import { AuthenticatedPageProps } from '~/components/AuthenticatedPage';

export const getDisplayName = (prefix: string, WrappedComponent: FC<AuthenticatedPageProps>) => `${prefix}(${WrappedComponent.displayName || WrappedComponent.name || 'Component'})`;
