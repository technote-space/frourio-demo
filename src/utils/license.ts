import type { License } from '~/types';
import licenses from '../../license.json';

export const getLicenseList = (): Array<License> => {
  return Object.values(licenses).map(license => ({
    name: license.name,
    version: license.version,
    licenses: typeof license.licenses === 'string' ? license.licenses : license.licenses.join(', '),
    repository: license.repository,
    licenseText: license.licenseText,
  }));
};
