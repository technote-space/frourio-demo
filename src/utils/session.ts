export const getSession = async <T>(key: string, generator: () => Promise<T | undefined>, setter?: (value: any) => void): Promise<T | undefined> => {
  const value = sessionStorage.getItem(key);
  if (null === value) {
    const result = await generator();
    if (undefined !== result && setter) {
      setter(result);
    }

    return result;
  }

  return JSON.parse(value);
};
export const setSession = (key: string, value: any): void => {
  sessionStorage.setItem(key, JSON.stringify(value));
};

// export const getAuthTokenCache   = async(generator: () => Promise<string | undefined>) => getSession(`authToken`, generator, setAuthTokenCache);
// export const setAuthTokenCache   = (value: string) => setSession(`authToken`, value);
// export const clearAuthTokenCache = () => sessionStorage.removeItem(`authToken`);
