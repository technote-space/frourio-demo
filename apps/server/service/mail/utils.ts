import type { MailOptions, MailSettings, MailAddress } from '$/types';
import type { Primitive } from '@frourio-demo/types';
import { htmlToText } from 'html-to-text';
import { fork } from 'child_process';
import { FRONT_URL, SMTP_BCC, SMTP_FROM, SMTP_HOST, SMTP_PASS, SMTP_PORT, SMTP_SECURE, SMTP_USER } from '$/service/env';
import { replaceVariables } from '@frourio-demo/utils/string';
import HeadTemplate from './templates/Head.html';
import HeaderTemplate from './templates/Header.html';
import FooterTemplate from './templates/Footer.html';

export const getSmtpOptions = (): MailOptions => ({
  host: SMTP_HOST,
  port: SMTP_PORT,
  secure: SMTP_SECURE,
  auth: {
    user: SMTP_USER,
    pass: SMTP_PASS,
  },
  tls: { rejectUnauthorized: false },
});
export const getText = (text: string | undefined, html: string): string => text ?? htmlToText(html);
export const getMailSettings = ({ to, bcc, subject, html, text }: {
  to: MailAddress;
  bcc?: MailAddress;
  subject: string;
  html: string;
  text?: string;
}): MailSettings => ({
  from: SMTP_FROM,
  to,
  bcc,
  subject,
  html,
  text: getText(text, html),
});

export const getTemplateVariables = () => ({
  'head': HeadTemplate,
  'header': HeaderTemplate,
  'footer': FooterTemplate,
});
export const getCommonVariables = () => ({
  'logo_url': `${FRONT_URL}/favicon.png`,
  'top_link': `${FRONT_URL}`,
  'terms_link': `${FRONT_URL}/terms`,
  'privacy_link': `${FRONT_URL}/privacy`,
  'contact_link': `${FRONT_URL}/contact`,
});

export const send = async(options: MailOptions, settings: MailSettings): Promise<boolean> => {
  const child = fork(
    'tasks/mail.js',
    [],
    { execArgv: ['-r', 'ts-node/register'] },
  );
  return child.send({ options, settings });
};

export const sendHtmlMail = async(to: MailAddress, subject: string, template: string, variables?: Record<string, Primitive>): Promise<boolean> => send(getSmtpOptions(), getMailSettings({
  to,
  bcc: SMTP_BCC,
  subject: `【Frourioの宿】${subject}`,
  html: replaceVariables(replaceVariables(template, getTemplateVariables()), {
    ...getCommonVariables(),
    ...variables,
  }),
}));
