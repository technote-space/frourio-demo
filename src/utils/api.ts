/* eslint-disable @typescript-eslint/no-explicit-any */
import type { AspidaResponse } from 'aspida';
import type { ApiInstance } from '$/api/$api';
import type { MaybeUndefined } from '~/types';
import aspida from '@aspida/axios';
import { AxiosError } from 'axios';
import { singular } from 'pluralize';
import api from '$/api/$api';
import { changePage, setError } from '~/utils/actions';

export const client = api(aspida());

export const isAxiosError = (target: any): target is AxiosError => {
  return typeof target === 'object' &&
    'config' in target &&
    'request' in target &&
    'response' in target;
};

export const handleAuthError = async <T, U, V, API extends (...args: Array<any>) => Promise<AspidaResponse<T, U, V>>>(
  dispatch: ((value: { type: string; [key: string]: any }) => void),
  fallback: T,
  api: API,
  ...option: Parameters<API>
): Promise<T | MaybeUndefined<T>> | never => {
  try {
    const result = await api(...option);
    console.log(result);

    return result.body;
  } catch (error) {
    console.log(error);
    if (isAxiosError(error)) {
      console.log(error.response);
      if (error.response && 401 === error.response.status) {
        setError(dispatch, error.response.data.message);
        changePage(dispatch, 'logout');
        if (!fallback) {
          return undefined as MaybeUndefined<T>;
        }

        return fallback;
      }

      if (error.response) {
        setError(dispatch, error.response.data.message);
        throw error;
      }
    }

    setError(dispatch, error.message);
    throw error;
  }
};

export type DataTableApi = {
  get: (option?: any) => Promise<AspidaResponse<any, any, any>>;
  post: (option?: any) => Promise<AspidaResponse<any, any, any>>;
  detail: (id: number) => {
    patch: (option?: any) => Promise<AspidaResponse<any, any, any>>;
    delete: (option?: any) => Promise<AspidaResponse<any, any, any>>;
  }
};
export type DataTableApiModels = {
  [key in keyof ApiInstance]: ApiInstance[key] extends {
    get,
    post,
  } ? key : never
}[keyof ApiInstance]
export const getDataTableApi = (model: DataTableApiModels): DataTableApi => ({
  ...client[model],
  detail: id => ({
    patch: client[model][`_${singular(model)}Id`](id).patch,
    delete: client[model][`_${singular(model)}Id`](id).delete,
  }),
});

export const processUpdateData = <T extends Partial<{
  updatedAt,
  createdAt,
}>>(data: T): Omit<T, 'updatedAt' | 'createdAt'> => {
  delete data.updatedAt;
  delete data.createdAt;
  return data;
};
