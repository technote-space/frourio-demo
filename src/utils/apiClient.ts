import aspida from '@aspida/axios';
import api from '$/api/$api';

export const apiClient = api(aspida());
