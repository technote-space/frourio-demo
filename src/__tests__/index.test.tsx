import dotenv from 'dotenv';
import Index from '~/pages/index';
import { render, setup, useNock, setToken, setInvalidToken, setDarkMode, act } from '~/__tests__/utils';
import user from '@testing-library/user-event';

dotenv.config({ path: 'server/.env' });

setup();

describe('Index', () => {
  it('should show login form', async() => {
    const { asFragment, getAllByText, getByText, getByLabelText } = render(<Index/>);

    expect(getAllByText('予約システム')).toHaveLength(2); // header, footer
    expect(getByText('Email address')).toBeVisible();
    expect(getByLabelText(/Email address/)).toBeVisible();
    expect(getByText('Password')).toBeVisible();
    expect(getByLabelText(/Password/)).toBeVisible();
    expect(getByText('Login')).toBeVisible();
    expect(asFragment()).toMatchSnapshot();
  });

  it('should fail to login', async() => {
    const login = jest.fn();
    useNock()
      .post('/login', body => {
        login(body);
        return body;
      }).reply(400);
    setInvalidToken();

    const { asFragment, getByText, getByLabelText, getByTestId, findByText } = render(<Index/>);

    user.type(getByLabelText(/Email address/), 'test@example.com');
    user.type(getByLabelText(/Password/), 'password');
    user.click(getByTestId('password-switch'));
    await act(async() => {
      user.click(getByText('Login'));
    });

    await findByText('Request failed with status code 400');

    expect(login).toBeCalledWith({ email: 'test@example.com', pass: 'password' });
    expect(asFragment()).toMatchSnapshot();
  });

  it('should fail to login (invalid header)', async() => {
    const login = jest.fn();
    useNock()
      .post('/login', body => {
        login(body);
        return body;
      }).reply(204);

    const { asFragment, getByText, getByLabelText, findByText } = render(<Index/>);

    user.type(getByLabelText(/Email address/), 'test@example.com');
    user.type(getByLabelText(/Password/), 'password');
    await act(async() => {
      user.click(getByText('Login'));
    });

    await findByText('Invalid header');

    expect(login).toBeCalledWith({ email: 'test@example.com', pass: 'password' });
    expect(asFragment()).toMatchSnapshot();
  });

  it('should success to login', async() => {
    const login = jest.fn();
    useNock()
      .post('/login', body => {
        login(body);
        return body;
      }).reply(204, undefined, {
        authorization: 'test-token',
      })
      .get('/admin').reply(200, { name: null, icon: 'http://localhost:8080/api/icons/dummy.svg' })
      .get('/dashboard/rooms').reply(200, [])
      .get(/\/dashboard\/(checkin|checkout)/).reply(200, {
        'data': [],
        'page': 0,
        'totalCount': 0,
      })
      .get(/\/dashboard\/sales/).reply(200, []);

    const { getByText, getByLabelText, findAllByText, getAllByText, container, findByText, findByTestId } = render(
      <Index/>,
    );

    user.type(getByLabelText(/Email address/), 'test@example.com');
    user.type(getByLabelText(/Password/), 'password');
    await act(async() => {
      user.click(getByText('Login'));
    });

    expect(await findAllByText('チェックイン')).toHaveLength(2); // table title, table header

    expect(login).toBeCalledWith({ email: 'test@example.com', pass: 'password' });
    expect(getAllByText('チェックアウト')).toHaveLength(2); // table title, table header
    expect(getByText('全部屋')).toBeVisible();

    // test header
    const buttons = container.querySelectorAll('header .MuiSvgIcon-root');
    expect(buttons).toHaveLength(2);
    user.click(getAllByText('予約システム')[0]); // header link
    user.click(buttons[1]); // toggle dark mode

    // test sidebar
    user.click(buttons[0]);
    await findByTestId('admin-avatar');
    const menuClose = await findByTestId('drawer-layer');
    user.click(menuClose);
  });

  it('should logout automatically', async() => {
    useNock()
      .get('/admin').reply(401, {
        message: 'test error',
      })
      .get('/dashboard/rooms').reply(200, [])
      .get(/\/dashboard\/(checkin|checkout)/).reply(200, {
        'data': [],
        'page': 0,
        'totalCount': 0,
      })
      .get(/\/dashboard\/sales/).reply(200, []);
    setToken('token');

    const { findByText } = render(<Index/>);

    await findByText('Login');
  });

  it('should login automatically and logout', async() => {
    useNock()
      .get('/admin').reply(200, { name: 'test name', icon: null })
      .get('/dashboard/rooms').reply(200, [])
      .get(/\/dashboard\/(checkin|checkout)/).reply(200, {
        'data': [],
        'page': 0,
        'totalCount': 0,
      })
      .get(/\/dashboard\/sales/).reply(200, []);
    setToken('token');
    setDarkMode(true);

    const { getByText, findByText, findByTestId, container } = render(<Index/>);

    await findByTestId('select-date');

    const buttons = container.querySelectorAll('header .MuiSvgIcon-root');
    user.click(buttons[0]);
    await findByText('test name');
    user.click(getByText('ログアウト'));
    await findByText('Login');
    expect(getByText('Email address')).toBeVisible();
    expect(getByText('Password')).toBeVisible();
  });

  it('should show license dialog', async() => {
    useNock()
      .get('/admin').reply(200, { name: 'test name', icon: null })
      .get('/dashboard/rooms').reply(200, [])
      .get(/\/dashboard\/(checkin|checkout)/).reply(200, {
        'data': [],
        'page': 0,
        'totalCount': 0,
      })
      .get(/\/dashboard\/sales/).reply(200, []);
    setToken('token');

    const { findByText, findByTestId, findAllByTestId, container } = render(<Index/>);

    await findByTestId('select-date');

    const buttons = container.querySelectorAll('header .MuiSvgIcon-root');
    user.click(buttons[0]);
    user.click(await findByText('ライセンス'));
    user.click((await findAllByTestId('license-item'))[0]);
    user.click(await findByTestId('close-license'));
    user.click(await findByTestId('close-license-list'));
  });
});
