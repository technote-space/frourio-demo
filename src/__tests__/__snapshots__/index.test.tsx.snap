// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index should fail to login (invalid header) 1`] = `
<DocumentFragment>
  <div
    class="MuiSnackbar-root MuiSnackbar-anchorOriginBottomCenter makeStyles-content-64"
  >
    <div
      aria-describedby="client-snackbar"
      class="MuiPaper-root MuiSnackbarContent-root makeStyles-warning-89 makeStyles-content-64 MuiPaper-elevation6"
      direction="up"
      role="alert"
      style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="MuiSnackbarContent-message"
      >
        <span
          class="makeStyles-message-92"
          id="client-snackbar"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-icon-90 makeStyles-iconVariant-91"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
            />
          </svg>
          Invalid header
        </span>
      </div>
      <div
        class="MuiSnackbarContent-action"
      >
        <button
          aria-label="close"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-icon-90"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    class="makeStyles-wrap-65"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-root-68 MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <h6
          class="MuiTypography-root makeStyles-title-70 MuiTypography-h6"
        >
          予約システム
        </h6>
        <div
          class="makeStyles-grow-71"
        />
        <button
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69zM12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6c3.31 0 6 2.69 6 6s-2.69 6-6 6z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
    <main
      class="makeStyles-main-66"
    >
      <div
        class="makeStyles-contents-67"
      >
        <div
          class="makeStyles-wrap-78"
          data-testid="page-dashboard"
        >
          <div
            class="makeStyles-wrap-79"
          >
            <div
              class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
            >
              <div
                class="MuiCardContent-root"
              >
                <form
                  action="#"
                >
                  <div
                    class="MuiFormControl-root makeStyles-input-82"
                  >
                    <label
                      class="MuiFormLabel-root MuiFormLabel-filled Mui-required"
                      for="email"
                    >
                      Email address
                      <span
                        aria-hidden="true"
                        class="MuiFormLabel-asterisk"
                      >
                         *
                      </span>
                    </label>
                    <div
                      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiInput-input"
                        id="email"
                        name="email"
                        placeholder="Enter email address"
                        required=""
                        type="email"
                        value="test@example.com"
                      />
                    </div>
                    <div
                      class="makeStyles-error-81"
                    />
                  </div>
                  <div
                    class="MuiFormControl-root makeStyles-input-82"
                  >
                    <label
                      class="MuiFormLabel-root MuiFormLabel-filled Mui-required"
                      for="password"
                    >
                      Password
                      <span
                        aria-hidden="true"
                        class="MuiFormLabel-asterisk"
                      >
                         *
                      </span>
                    </label>
                    <div
                      class="makeStyles-wrap-83"
                    >
                      <div
                        class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-input-84 MuiInputBase-formControl MuiInput-formControl"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiInputBase-input MuiInput-input"
                          id="password"
                          name="password"
                          placeholder="Enter password"
                          required=""
                          type="password"
                          value="password"
                        />
                      </div>
                      <div
                        class="makeStyles-toggle-85"
                      >
                        <button
                          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
                          data-testid="password-switch"
                          tabindex="0"
                          type="button"
                        >
                          <span
                            class="MuiButton-label"
                          >
                            Show
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </div>
                    <div
                      class="makeStyles-error-81"
                    />
                  </div>
                  <div
                    class="makeStyles-login-80"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                      tabindex="0"
                      type="submit"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        Login
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
    <footer
      class="makeStyles-footer-76"
    >
      <div
        class="makeStyles-wrap-77"
      >
        2021 — 
        <strong>
          予約システム
        </strong>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;

exports[`Index should fail to login 1`] = `
<DocumentFragment>
  <div
    class="MuiSnackbar-root MuiSnackbar-anchorOriginBottomCenter makeStyles-content-31"
  >
    <div
      aria-describedby="client-snackbar"
      class="MuiPaper-root MuiSnackbarContent-root makeStyles-warning-56 makeStyles-content-31 MuiPaper-elevation6"
      direction="up"
      role="alert"
      style="opacity: 1; transform: scale(1, 1); transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="MuiSnackbarContent-message"
      >
        <span
          class="makeStyles-message-59"
          id="client-snackbar"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root makeStyles-icon-57 makeStyles-iconVariant-58"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M1 21h22L12 2 1 21zm12-3h-2v-2h2v2zm0-4h-2v-4h2v4z"
            />
          </svg>
          Request failed with status code 400
        </span>
      </div>
      <div
        class="MuiSnackbarContent-action"
      >
        <button
          aria-label="close"
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root makeStyles-icon-57"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    class="makeStyles-wrap-32"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-root-35 MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <h6
          class="MuiTypography-root makeStyles-title-37 MuiTypography-h6"
        >
          予約システム
        </h6>
        <div
          class="makeStyles-grow-38"
        />
        <button
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69zM12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6c3.31 0 6 2.69 6 6s-2.69 6-6 6z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
    <main
      class="makeStyles-main-33"
    >
      <div
        class="makeStyles-contents-34"
      >
        <div
          class="makeStyles-wrap-45"
          data-testid="page-dashboard"
        >
          <div
            class="makeStyles-wrap-46"
          >
            <div
              class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
            >
              <div
                class="MuiCardContent-root"
              >
                <form
                  action="#"
                >
                  <div
                    class="MuiFormControl-root makeStyles-input-49"
                  >
                    <label
                      class="MuiFormLabel-root MuiFormLabel-filled Mui-required"
                      for="email"
                    >
                      Email address
                      <span
                        aria-hidden="true"
                        class="MuiFormLabel-asterisk"
                      >
                         *
                      </span>
                    </label>
                    <div
                      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiInput-input"
                        id="email"
                        name="email"
                        placeholder="Enter email address"
                        required=""
                        type="email"
                        value="test@example.com"
                      />
                    </div>
                    <div
                      class="makeStyles-error-48"
                    />
                  </div>
                  <div
                    class="MuiFormControl-root makeStyles-input-49"
                  >
                    <label
                      class="MuiFormLabel-root MuiFormLabel-filled Mui-required"
                      for="password"
                    >
                      Password
                      <span
                        aria-hidden="true"
                        class="MuiFormLabel-asterisk"
                      >
                         *
                      </span>
                    </label>
                    <div
                      class="makeStyles-wrap-50"
                    >
                      <div
                        class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-input-51 MuiInputBase-formControl MuiInput-formControl"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiInputBase-input MuiInput-input"
                          id="password"
                          name="password"
                          placeholder="Enter password"
                          required=""
                          type="text"
                          value="password"
                        />
                      </div>
                      <div
                        class="makeStyles-toggle-52"
                      >
                        <button
                          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
                          data-testid="password-switch"
                          tabindex="0"
                          type="button"
                        >
                          <span
                            class="MuiButton-label"
                          >
                            Hide
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </div>
                    <div
                      class="makeStyles-error-48"
                    />
                  </div>
                  <div
                    class="makeStyles-login-47"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                      tabindex="0"
                      type="submit"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        Login
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
    <footer
      class="makeStyles-footer-43"
    >
      <div
        class="makeStyles-wrap-44"
      >
        2021 — 
        <strong>
          予約システム
        </strong>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;

exports[`Index should show login form 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-wrap-6"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary makeStyles-root-9 MuiPaper-elevation4"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <h6
          class="MuiTypography-root makeStyles-title-11 MuiTypography-h6"
        >
          予約システム
        </h6>
        <div
          class="makeStyles-grow-12"
        />
        <button
          class="MuiButtonBase-root MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69zM12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6c3.31 0 6 2.69 6 6s-2.69 6-6 6z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
    <main
      class="makeStyles-main-7"
    >
      <div
        class="makeStyles-contents-8"
      >
        <div
          class="makeStyles-wrap-19"
          data-testid="page-dashboard"
        >
          <div
            class="makeStyles-wrap-20"
          >
            <div
              class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
            >
              <div
                class="MuiCardContent-root"
              >
                <form
                  action="#"
                >
                  <div
                    class="MuiFormControl-root makeStyles-input-23"
                  >
                    <label
                      class="MuiFormLabel-root Mui-required"
                      for="email"
                    >
                      Email address
                      <span
                        aria-hidden="true"
                        class="MuiFormLabel-asterisk"
                      >
                         *
                      </span>
                    </label>
                    <div
                      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiInput-input"
                        id="email"
                        name="email"
                        placeholder="Enter email address"
                        required=""
                        type="email"
                        value=""
                      />
                    </div>
                    <div
                      class="makeStyles-error-22"
                    />
                  </div>
                  <div
                    class="MuiFormControl-root makeStyles-input-23"
                  >
                    <label
                      class="MuiFormLabel-root Mui-required"
                      for="password"
                    >
                      Password
                      <span
                        aria-hidden="true"
                        class="MuiFormLabel-asterisk"
                      >
                         *
                      </span>
                    </label>
                    <div
                      class="makeStyles-wrap-24"
                    >
                      <div
                        class="MuiInputBase-root MuiInput-root MuiInput-underline makeStyles-input-25 MuiInputBase-formControl MuiInput-formControl"
                      >
                        <input
                          aria-invalid="false"
                          class="MuiInputBase-input MuiInput-input"
                          id="password"
                          name="password"
                          placeholder="Enter password"
                          required=""
                          type="password"
                          value=""
                        />
                      </div>
                      <div
                        class="makeStyles-toggle-26"
                      >
                        <button
                          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeSmall MuiButton-sizeSmall"
                          data-testid="password-switch"
                          tabindex="0"
                          type="button"
                        >
                          <span
                            class="MuiButton-label"
                          >
                            Show
                          </span>
                          <span
                            class="MuiTouchRipple-root"
                          />
                        </button>
                      </div>
                    </div>
                    <div
                      class="makeStyles-error-22"
                    />
                  </div>
                  <div
                    class="makeStyles-login-21"
                  >
                    <button
                      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
                      tabindex="0"
                      type="submit"
                    >
                      <span
                        class="MuiButton-label"
                      >
                        Login
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                </form>
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
    <footer
      class="makeStyles-footer-17"
    >
      <div
        class="makeStyles-wrap-18"
      >
        2021 — 
        <strong>
          予約システム
        </strong>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
